### PR TITLE
UNR-1754 Auto deploy config

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added a setting to disable auto-starting a local deployment when the user presses play.

How can this be verified by QA?: Try disabling the "Auto-start local deployment" setting and see if a deployment auto starts.

#### Primary reviewers
@Vatyx @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -1218,7 +1218,8 @@ void USpatialReceiver::ProcessRPCEventField(Worker_EntityId EntityId, const Work
 			UFunction* Function = ClassInfo.RPCs[Payload.Index];
 			const FRPCInfo& RPCInfo = ClassInfoManager->GetRPCInfo(TargetObject, Function);
 
-			if (!IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, RPCInfo.Type))
+			if (!IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, RPCInfo.Type)
+				&& !IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, ESchemaComponentType::SCHEMA_Invalid))
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: This doesn't look like your change :P

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 2]
```diff
@@ -1656,7 +1657,8 @@ void USpatialReceiver::ProcessQueuedActorRPCsOnEntityCreation(AActor* Actor, RPC
 		const FUnrealObjectRef ObjectRef = PackageMap->GetUnrealObjectRefFromObject(Actor);
 		check(ObjectRef != FUnrealObjectRef::UNRESOLVED_OBJECT_REF);
 
-		if (!IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, RPCInfo.Type))
+		if (!IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, RPCInfo.Type)
+			&& !IncomingRPCs.ObjectHasRPCsQueuedOfType(ObjectRef.Entity, ESchemaComponentType::SCHEMA_Invalid))
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: And this doesn't look like your change as well.

**Comment by joshuahuburn** [comment]
Also UNR-1745

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 3]
```diff
@@ -252,6 +252,10 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
 	UPROPERTY(EditAnywhere, config, Category = "Launch", meta = (ConfigRestartRequired = false, DisplayName = "Stop on exit"))
 	bool bStopSpatialOnExit;
 
+	/** Start a local SpatialOS deployment when clicking 'Play'. */
+	UPROPERTY(EditAnywhere, config, Category = "Launch", meta = (ConfigRestartRequired = false, DisplayName = "Auto-start local deployment"))
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]: Is it intentional that the names are different: "Auto-start local deployment" and bAutoStartSpatialDeployment?

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 4]
```diff
@@ -693,6 +694,22 @@ void FSpatialGDKEditorToolbarModule::OnPropertyChanged(UObject* ObjectBeingModif
 		{
 			bStopSpatialOnExit = Settings->bStopSpatialOnExit;
 		}
+		else if (PropertyName.ToString() == TEXT("bAutoStartSpatialDeployment"))
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Told you it's a trap :D

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h (hunk 3 above)
Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]: Renamed to bAutoStartLocalDeployment

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp (hunk 4 above)
Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Baahahaahahahah no way

**Pull request merged**